astropy/io/fits/tests/test_fitsheader.py

```diff
@@ -77,3 +77,34 @@ def test_table(self, capsys):
         assert out[1].endswith('|   0 |   NAXIS |     3 |')
         assert out[2].endswith('|   0 |   NAXIS |     0 |')
         assert out[3].endswith('|   0 |   NAXIS |     2 |')
+
+    def test_fitsort(self, capsys):
+        fitsheader.main(['-e', '0', '-f', '-k', 'EXPSTART', '-k', 'EXPTIME',
+                         self.data('test0.fits'), self.data('test1.fits')])
+        out, err = capsys.readouterr()
+        out = out.splitlines()
+        assert len(out) == 5
+        assert out[3].endswith('test0.fits 49491.65366175    0.23')
+        assert out[4].endswith('test1.fits 49492.65366175    0.22')
+
+        fitsheader.main(['-e', '0', '-f', '-k', 'EXPSTART', '-k', 'EXPTIME',
+                         self.data('test0.fits')])
+        out, err = capsys.readouterr()
+        out = out.splitlines()
+        assert len(out) == 4
+        assert out[3].endswith('test0.fits 49491.65366175    0.23')
+
+    def test_dotkeyword(self, capsys):
+        fitsheader.main(['-e', '0', '-k', 'ESO DET ID',
+                         self.data('fixed-1890.fits')])
+        out, err = capsys.readouterr()
+        out = out.splitlines()
+        assert len(out) == 2
+        assert out[1].strip().endswith("HIERARCH ESO DET ID = 'DV13' / Detector system Id")
+
+        fitsheader.main(['-e', '0', '-k', 'ESO.DET.ID',
+                         self.data('fixed-1890.fits')])
+        out, err = capsys.readouterr()
+        out = out.splitlines()
+        assert len(out) == 2
+        assert out[1].strip().endswith("HIERARCH ESO DET ID = 'DV13' / Detector system Id")
```

Comment:
Could you also add a test for the table header line with multiple HDUs ? E.g. testing something like this:
```
❯ fitsheader -f -k NAXIS astropy/io/fits/tests/data/t*
               filename               0:NAXIS 1:NAXIS 2:NAXIS 3:NAXIS 4:NAXIS
                str37                   str1    str1    str1    str1    str1 
------------------------------------- ------- ------- ------- ------- -------
astropy/io/fits/tests/data/table.fits       0       2      --      --      --
   astropy/io/fits/tests/data/tb.fits       0       2      --      --      --
 astropy/io/fits/tests/data/tdim.fits       0       2      --      --      --
astropy/io/fits/tests/data/test0.fits       0       2       2       2       2
astropy/io/fits/tests/data/test1.fits       0       2       2       2       2
```

Reply:
Done!